src/deadline_worker_agent/sessions/actions/run_attachment_download.py

```diff
@@ -0,0 +1,338 @@
+
+        # Generate absolute Path Mapping to local session (no storage profile)
+        # returns root path to PathMappingRule mapping
+        dynamic_mapping_rules: dict[str, PathMappingRule] = (
+            session._asset_sync.generate_dynamic_path_mapping(
+                session_dir=session._session.working_directory,
+                attachments=attachments,
+            )
+        )
+
+        # Aggregate manifests (with step step dependency handling)
+        merged_manifests_by_root: dict[str, BaseAssetManifest] = (
+            session._asset_sync._aggregate_asset_root_manifests(
+                session_dir=session._session.working_directory,
+                s3_settings=s3_settings,
+                queue_id=session._queue_id,
+                job_id=session._queue._job_id,
+                attachments=attachments,
+                step_dependencies=step_dependencies,
+                dynamic_mapping_rules=dynamic_mapping_rules,
+                storage_profiles_path_mapping_rules=storage_profiles_path_mapping_rules_dict,
+            )
+        )
+
+        vfs_handled = self._vfs_handling(
+            session=session,
+            attachments=attachments,
+            merged_manifests_by_root=merged_manifests_by_root,
+            s3_settings=s3_settings,
+        )
+
+        if not vfs_handled:
+            job_attachment_path_mappings = [asdict(r) for r in dynamic_mapping_rules.values()]
+
+            # Open Job Description session implementation details -- path mappings are sorted.
+            # bisect.insort only supports the 'key' arg in 3.10 or later, so
+            # we first extend the list and sort it afterwards.
+            if session._session._path_mapping_rules:
+                session._session._path_mapping_rules.extend(job_attachment_path_mappings)
+            else:
+                session._session._path_mapping_rules = [
+                    OpenjdPathMapping.from_dict(r) for r in job_attachment_path_mappings
+                ]
+
+            # Open Job Description Sessions sort the path mapping rules based on length of the parts make
+            # rules that are subsets of each other behave in a predictable manner. We must
+            # sort here since we're modifying that internal list appending to the list.
+            session._session._path_mapping_rules.sort(key=lambda rule: -len(rule.source_path.parts))
+
+            # =========================== TO BE DELETED ===========================
+            path_mapping_file_path: str = os.path.join(
+                session._session.working_directory, "path_mapping"
+            )
+            for rule in job_attachment_path_mappings:
+                rule["source_path"] = rule["destination_path"]
+
+            with open(path_mapping_file_path, "w", encoding="utf8") as f:
+                f.write(json.dumps([rule for rule in job_attachment_path_mappings]))
+            # =========================== TO BE DELETED ===========================
```

Comment:
Keeping it for now, pending changes and testing with attachment upload cli action integration.

Reply:
I actually think writing out a path mapping file is super useful for users? They can easily cat this file and use `jq` Or are we depending on the environment variable for path mapping for customers?

Reply:
Openjd session already writes the path mapping files, it's in a slightly different format tho, that's why I was thinking whether still keep this path mapping file or reuse the path written by openjd. Openjd would need to change to return the file path, and the attachment CLI would need slight change to read the following file.

```
{
    "version": "pathmapping-1.0",
    "path_mapping_rules": [{
        "source_path_format": "POSIX",
        "source_path": "/local/bea/bealine-clients/deadline-cloud-worker-agent/scripts/submit_jobs/asset_example",
        "destination_path": "/sessions/session-39d74575600f47eb80d31d440721a27b7mrqf9ip/assetroot-a7714e87e776d9f1c179"
    }]
}
```